x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/revert_prebuilt_rule/revert_prebuilt_rule_handler.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IKibanaResponse, KibanaRequest, KibanaResponseFactory } from '@kbn/core/server';
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { getErrorMessage, getErrorStatusCode } from '../../../../../utils/error_helpers';
+import type { RuleResponse } from '../../../../../../common/api/detection_engine/model/rule_schema';
+import type { BulkActionReversionSkipResult } from '../../../../../../common/api/detection_engine/prebuilt_rules';
+import {
+  type RevertPrebuiltRulesRequest,
+  type RevertPrebuiltRulesResponseBody,
+} from '../../../../../../common/api/detection_engine/prebuilt_rules';
+import type { SecuritySolutionRequestHandlerContext } from '../../../../../types';
+import { buildSiemResponse } from '../../../routes/utils';
+import { createPrebuiltRuleAssetsClient } from '../../logic/rule_assets/prebuilt_rule_assets_client';
+import {
+  normalizeErrorResponse,
+  type BulkActionError,
+} from '../../../rule_management/api/rules/bulk_actions/bulk_actions_response';
+import { fetchRulesByQueryOrIds } from '../../../rule_management/api/rules/bulk_actions/fetch_rules_by_query_or_ids';
+import { convertAlertingRuleToRuleResponse } from '../../../rule_management/logic/detection_rules_client/converters/convert_alerting_rule_to_rule_response';
+import type { RuleTriad } from '../../model/rule_groups/get_rule_groups';
+import { zipRuleVersions } from '../../logic/rule_versions/zip_rule_versions';
+import { mergeAndUpdatePrebuiltRules } from '../../logic/rule_objects/merge_and_update_prebuilt_rules';
+import { getConcurrencyErrors } from './get_concurrrency_errors';
+import { filterRulesToRevert } from './filter_rules_to_revert';
+
+const MAX_RULES_TO_REVERT = 1;
+
+export const revertPrebuiltRuleHandler = async (
+  context: SecuritySolutionRequestHandlerContext,
+  request: KibanaRequest<undefined, undefined, RevertPrebuiltRulesRequest>,
+  response: KibanaResponseFactory
+) => {
+  const siemResponse = buildSiemResponse(response);
+  const { id, revision, version } = request.body;
+
+  try {
+    const ctx = await context.resolve(['core', 'alerting', 'securitySolution']);
+    const soClient = ctx.core.savedObjects.client;
+    const rulesClient = await ctx.alerting.getRulesClient();
+    const detectionRulesClient = ctx.securitySolution.getDetectionRulesClient();
+    const ruleAssetsClient = createPrebuiltRuleAssetsClient(soClient);
+
+    const concurrencySet = { [id]: { revision, version } };
+
+    const updated: RuleResponse[] = [];
+    const errors: BulkActionError[] = [];
+
+    const fetchRulesOutcome = await fetchRulesByQueryOrIds({
+      rulesClient,
+      query: undefined,
+      ids: [id],
+      maxRules: MAX_RULES_TO_REVERT,
+    });
```

Comment:
Until we revert in bulk it's possible to read a single rule via `rulesClient.getRule()`.

Reply:
Changed